src/CosyComposer.php

```diff
@@ -647,6 +647,12 @@ public function run()
         $direct = null;
         if ($config->shouldCheckDirectOnly()) {
             $this->log('Checking only direct dependencies since config option check_only_direct_dependencies is enabled');
+            $extends_responsible = $config->getExtendNameForKey('check_only_direct_dependencies');
+            if ($extends_responsible) {
```

Comment:
This should be a helper function so we can log consistent messages about overrides